testhelpers/chain.go

```diff
@@ -251,3 +260,13 @@ func (bs *FakeBlockProvider) NewBlockWithMessages(nonce uint64, messages []*type
 func (bs *FakeBlockProvider) NewBlock(nonce uint64, parents ...*types.Block) *types.Block {
 	return bs.NewBlockWithMessages(nonce, []*types.SignedMessage{}, parents...)
 }
+
+// NewBlkTimeForTestGetter returns a closure that returns a uint64 unique to that invocation.
+// The uint64 is unique wrt the closure returned, not globally.
+func NewBlkTimeForTestGetter() func() uint64 {
+	var ticker uint64
+	return func() uint64 {
+		ticker++
+		return ticker
+	}
+}
```

Comment:
I found the use of this really confusing, for probably a few reasons:
- there are no units specified anywhere. I guess it's intended to be seconds?
- neither a starting timestamp nor the block time are parameters
- the name / documentation doesn't describe intention

I *think* this is intended as a thing that generates a sequence of block timestamps, such that if sequential blocks use sequential values, their timestamps will be deemed valid. I think it's just a fluke that the implicit increment "1" matches a block time that we set for tests, but it's a huge smell to me that we can't (or don't) set a value of "30" (and still have tests run nearly instantly).

If that's right, then I think this should be called something like `NewBlockTimeTicker` and take at least  parameters, being a start timestamp and a tick duration. Possibly we want a type alias for `BlockTimestamp` so that the `uint64` makes sense out of context.

It would be more direct as struct with a method than a closure, too. The `func() uint64` when this type is referred to from elsewhere is pretty opaque. This would then let it be manipulated for tests, e.g. to generate an invalid timestamp.

Reply:
Agreed, this needs a better comment stating the intention.

> I think it's just a fluke that the implicit increment "1" matches a block time that we set for tests, but it's a huge smell to me that we can't (or don't) set a value of "30" (and still have tests run nearly instantly).

This is being used by unit tests, they don't perform any mining. This won't work for tests that do mining moving forward. I will sketch something out like the `NewBlockTimeTicker` you've described.